python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1778,6 +1779,52 @@ def get_event_records(
         """
         return self._event_storage.get_event_records(event_records_filter, limit, ascending)
 
+    @public
+    @traced
+    def get_status_by_partition(
+        self, asset_key: AssetKey, partitions_def: PartitionsDefinition, partition_keys: List[str]
+    ) -> Dict:
+        """Get the current status of provided partition_keys.
+
+        Args:
+            asset_key (AssetKey):
+            partitions_def (PartitionsDefinition):
+            partition_keys (List[str]):
+
+        Returns:
+            Dict: status for each partition key
+
+        """
+        from dagster._core.storage.partition_status_cache import (
+            PartitionStatus,
+            get_and_update_asset_status_cache_value,
+        )
+
+        cached_value = get_and_update_asset_status_cache_value(self, asset_key, partitions_def)
+        materialized_partitions = cached_value.deserialize_materialized_partition_subsets(
+            partitions_def
+        ).get_partition_keys()
```

Comment:
I think you can avoid invoking `get_partition_keys` here. The `PartitionsSubset` objects that are returned by `deserialize_materialized_partition_subsets` support `in`, and it will be more performant than iterating through a list of partition keys.